### PR TITLE
Forest Stage and related actors and objects

### PR DESCRIPTION
**Pull request by petrie911**
En_Dnt_Demo, En_Dnt_Nomal, En_Dnt_Jiji, object_dns, object_dnk, object_hintnuts

This is way more complex than it needs to be. Also updated En_Hintnuts and En_Nutsball to use the new object file.

### CODE REVIEW

File: assets/xml/objects/object_dnk.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        <Animation Name="gDntStageHideAnim" Offset="0x430"/>
+        <Animation Name="gDntStageDanceAnim" Offset="0x6CC"/>
+        <Animation Name="gDntStageUnburrowAnim" Offset="0x894"/>
+        <Animation Name="gDntStageUpAnim" Offset="0xB70"/>
+        <Animation Name="gDntStageWalkAnim" Offset="0x2E1C"/>
+
+        <!-- Forest Stage scrub eye textures -->
+        <Texture Name="gDntStageEyeOpenTex" Format="rgb5a1" Width="8" Height="8" Offset="0x27D0"/>
+        <Texture Name="gDntStageEyeHalfTex" Format="rgb5a1" Width="8" Height="8" Offset="0x25D0"/>
+        <Texture Name="gDntStageEyeShutTex" Format="rgb5a1" Width="8" Height="8" Offset="0x2750"/>
```
Comment (fig02) [inline assets/xml/objects/object_dnk.xml:17]: OutName

File: src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.h [hunk 2]
```diff
@@ -6,11 +6,64 @@
...
+    /* 0x0154 */ s16 unkTimer2;
+    /* 0x0156 */ s16 debugArrowTimer;
+    /* 0x0158 */ s16 leaderSignal;
+    /* 0x015A */ s16 action;
+    /* 0x015C */ s16 prize;
+    /* 0x015E */ s16 subCamera;
+    /* 0x0160 */ Vec3f scrubPos[9];
+    /* 0x01CC */ Vec3f leaderPos;
+    /* 0x01D8 */ struct EnDntNomal* scrubs[9];
+    /* 0x01FC */ struct EnDntJiji* leader;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.h:25]: struct keyword needed? should be able to just import their headers

File: src/overlays/actors/ovl_En_Dnt_Jiji/z_en_dnt_jiji.h [hunk 3]
```diff
@@ -6,9 +6,30 @@
...
+    /* 0x0244 */ s16 blinkTimer;
+    /* 0x0246 */ s16 unkTimer;
+    /* 0x0248 */ s16 endFrame;
+    /* 0x024A */ s16 unburrow;
+    /* 0x024C */ s16 eyeState;
+    /* 0x024E */ s16 stageSignal;
+    /* 0x0250 */ s16 unused; // always set to 5
+    /* 0x0252 */ s16 attackFlag; // Is never set
+    /* 0x0254 */ s32 getItemId;
+    /* 0x0258 */ struct EnDntDemo* stage;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Dnt_Jiji/z_en_dnt_jiji.h:31]: same here

File: src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.h (hunk 2 above)
Comment (fig02) [reply src/overlays/actors/ovl_En_Dnt_Demo/z_en_dnt_demo.h:25]: nevermind

**Review by fig02 — APPROVED** [review]
changes made

File: src/overlays/actors/ovl_En_Dnt_Jiji/z_en_dnt_jiji.c [hunk 4]
```diff
@@ -51,53 +72,376 @@ static ColliderCylinderInit D_809F2FA0 = {
...
+            osSyncPrintf("棒 \n");
+            osSyncPrintf("棒 \n");
+            osSyncPrintf("棒 \n");
+            osSyncPrintf("棒 \n");
+            osSyncPrintf("棒 \n");
+            osSyncPrintf("棒 \n");
+            gSaveContext.itemGetInf[1] |= 0x4000;
+        }
+        this->actor.textId = 0;
+        if ((this->stage != 0) && (this->stage->actor.update != 0)) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dnt_Jiji/z_en_dnt_jiji.c:300]:
```suggestion
        if ((this->stage != NULL) && (this->stage->actor.update != NULL)) {
```

File: src/overlays/actors/ovl_En_Dnt_Nomal/z_en_dnt_nomal.c [hunk 5]
```diff
@@ -36,120 +70,823 @@ const ActorInit En_Dnt_Nomal_InitVars = {
...
+            this->eyeState = 0;
+            this->blinkTimer = (s16)Rand_ZeroFloat(60.0f) + 20;
+        }
+    }
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 20.0f, 60.0f, 0x1D);
+    if (this->type == ENDNTNOMAL_TARGET) {
+        Collider_SetQuadVertices(&this->targetQuad, &this->targetVtx[0], &this->targetVtx[1], &this->targetVtx[2],
+                      &this->targetVtx[3]);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dnt_Nomal/z_en_dnt_nomal.c:817]: The formatter needs to be run on this file (and maybe others)

File: src/overlays/actors/ovl_En_Dnt_Nomal/z_en_dnt_nomal.c [hunk 6]
```diff
@@ -36,120 +70,823 @@ const ActorInit En_Dnt_Nomal_InitVars = {
...
+s32 EnDntNomal_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                void* thisx) {
+    EnDntNomal* this = THIS;
+
+    if ((limbIndex == 1) || (limbIndex == 3) || (limbIndex == 4) || (limbIndex == 5) || (limbIndex == 6)) {
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_dnt_nomal.c", 1733);
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetEnvColor(POLY_OPA_DISP++, sLeafColors[this->type - ENDNTNOMAL_STAGE].r,
+                       sLeafColors[this->type - ENDNTNOMAL_STAGE].g, sLeafColors[this->type - ENDNTNOMAL_STAGE].b, 255);
+        CLOSE_DISPS(globalCtx->state.gfxCtx, (const char*)"../z_en_dnt_nomal.c", 1743);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dnt_Nomal/z_en_dnt_nomal.c:836]:
```suggestion
        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_dnt_nomal.c", 1743);
```

**Comment by fig02** [comment]
formatter ran

**Review by Roman971 — APPROVED** [review]

**Pull request merged**